pkgs/development/libraries/science/math/nccl/default.nix

```diff
@@ -17,16 +15,25 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" ];
 
-  nativeBuildInputs = [ which addOpenGLRunpath ];
+  nativeBuildInputs = [
+    which
+    addOpenGLRunpath
+    cuda_nvcc
+  ];
 
-  buildInputs = [ cudatoolkit ];
+  buildInputs = [
+    cuda_cudart
+  ];
 
   preConfigure = ''
     patchShebangs src/collectives/device/gen_rules.sh
+    export NVCC_APPEND_FLAGS+=' --compiler-bindir=${cudatoolkit.cc}/bin'
```

Comment:
This is to workaround nccl appending `-ccbin $(CXX)`: https://github.com/NVIDIA/nccl/blob/f3d51667838f7542df8ea32ea4e144d812b3ed7c/makefiles/common.mk#L65
Our `cuda_nvcc` setup-hook uses `NVCC_PREPEND_FLAGS`. NVCC ends up using the last `-ccbin` specified on the command line

Reply:
might be worth adding a comment? this is quite subtle